plugin/kubernetes/kubernetes.go

```diff
@@ -244,6 +245,17 @@ func (k *Kubernetes) InitKubeCache(ctx context.Context) (err error) {
 
 	k.opts.zones = k.Zones
 	k.opts.endpointNameMode = k.endpointNameMode
+	// Enable use of endpoint slices if the API supports the discovery v1 beta1 api
+	if _, err := kubeClient.Discovery().ServerResourcesForGroupVersion(discovery.SchemeGroupVersion.String()); err == nil {
+		k.opts.useEndpointSlices = true
+	}
+	// Disable use of endpoint slices for k8s versions 1.17 and 1.18. Endpoint slices were
+	// introduced in 1.17 but not fully integrated and enabled by default until k8s 1.19.
+	sv, _ := kubeClient.ServerVersion()
+	if strings.HasPrefix(sv.String(), "v1.18") || strings.HasPrefix(sv.String(), "v1.17") {
```

Comment:
hasn't the version stuff of k8s a whole api to check this? Meaning sv.Major() and friends exist and can be checked here?